Robal/src/main/java/ac/project/Robal/services/AccountService.java

```diff
@@ -1,5 +1,334 @@
+			Customer dbCustomer = customerRepository.findById(customer.getAccountId())
+					.orElseThrow(accountNotFound("Customer"));
+
+			dbCustomer.setName(customer.getName());
+			dbCustomer.setBillingAddress(customer.getBillingAddress());
+			dbCustomer.setEmail(customer.getEmail());
+			dbCustomer.setPassword(bCryptPasswordEncoder.encode(customer.getPassword()));
+			dbCustomer.setRole(customer.getRole());
+			dbCustomer.setPaymentMethod(customer.getPaymentMethod());
+
+			for (Order order : customer.getOrders()) {
+				if (!dbCustomer.getOrders().contains(order)) {
+					dbCustomer.getOrders().add(order);
+				}
+			}
+			return customerRepository.save(dbCustomer);
+		}
+	}
+
+	public Customer saveCustomerName(Customer customer) throws Exception, NotFoundException {
+
+		String name = customer.getName();
+		customer = customerRepository.findById(customer.getAccountId())
+				.orElse(null)/* .orElseThrow(accountNotFound("Customer")) */;
+		if (customer != null) {
+			customer.setName(name);
+
+		}
+		logger.info("***Repo:update customer name attemped for: " + customer.getEmail() + "***");
+		return saveCustomer(customer);
+
+	}
+
+	public Customer saveCustomerEmail(Customer customer) throws Exception, NotFoundException {
+
+		String email = customer.getEmail();
+
+		if (emailAlreadyExists(email)) {
+			throw new Exception("Email already taken.");
+		}
+
+		customer = customerRepository.findById(customer.getAccountId()).orElseThrow(accountNotFound("Customer"));
+		customer.setEmail(email);
+		logger.info("***Repo:update customer email attemped for: " + customer.getEmail() + "***");
+		return saveCustomer(customer);
+
+	}
+
+	public Customer saveCustomerPassword(Customer customer) throws Exception, NotFoundException {
+
+		String password = customer.getPassword();
+		customer = customerRepository.findById(customer.getAccountId()).orElseThrow(accountNotFound("Customer"));
+		customer.setPassword(bCryptPasswordEncoder.encode(password));
+		logger.info("***Repo:saveCustomerPassword attemped for: " + customer.getEmail() + "***");
+		return saveCustomer(customer);
+
+	}
+
+	public Customer saveCustomerBillingAddress(Customer customer) throws Exception, NotFoundException {
+
+		String billingAddress = customer.getBillingAddress();
+		customer = customerRepository.findById(customer.getAccountId()).orElseThrow(accountNotFound("Customer"));
+		customer.setName(billingAddress);
+		logger.info("***Repo:saveCustomerBillingAddress for: " + customer.getEmail() + "***");
+		return saveCustomer(customer);
+
+	}
+
+	public Customer saveCustomerPaymentMethod(Customer customer) throws Exception, NotFoundException {
+
+		String paymentMethod = customer.getPaymentMethod();
+		customer = customerRepository.findById(customer.getAccountId()).orElseThrow(accountNotFound("Customer"));
+		customer.setName(paymentMethod);
+		logger.info("***saveCustomerPaymentMethod for: " + customer.getEmail() + "***");
+		return saveCustomer(customer);
+
+	}

```

Comment:
All these methods saving individual fields can be combined into one giant save method for the whole object (-0.25%)